kld/src/cli/main.rs

```diff
@@ -118,37 +126,64 @@ fn main() {
-            announce,
-            fee_rate,
-        } => api.open_channel(public_key, satoshis, push_msat, announce, fee_rate)?,
-        Command::SetChannelFee {
-            id,
-            base_fee,
-            ppm_fee,
-        } => api.set_channel_fee(id, base_fee, ppm_fee)?,
-        Command::CloseChannel { id } => api.close_channel(id)?,
-        Command::NetworkNodes { id } => api.list_network_nodes(id)?,
-        Command::NetworkChannels { id } => api.list_network_channels(id)?,
+        Command::GetInfo
+        | Command::GetBalance
+        | Command::NewAddress
+        | Command::Withdraw { .. }
+        | Command::ListChannels
+        | Command::ListPeers
+        | Command::ConnectPeer { .. }
+        | Command::DisconnectPeer { .. }
+        | Command::OpenChannel { .. }
+        | Command::SetChannelFee { .. }
+        | Command::CloseChannel { .. }
+        | Command::NetworkNodes { .. }
+        | Command::NetworkChannels { .. } => {
+            if let (Some(target), Some(cert_path), Some(macaroon_path)) =
+                (&args.target, &args.cert_path, &args.macaroon_path)
+            {
+                let api = Api::new(target, cert_path, macaroon_path)?;
+                match args.command {
+                    Command::GetInfo => api.get_info()?,
+                    Command::GetBalance => api.get_balance()?,
+                    Command::NewAddress => api.new_address()?,
+                    Command::Withdraw {
+                        address,
+                        satoshis,
+                        fee_rate,
+                    } => api.withdraw(address, satoshis, fee_rate)?,
+                    Command::ListChannels => api.list_channels()?,
+                    Command::ListPeers => api.list_peers()?,
+                    Command::ConnectPeer { public_key } => api.connect_peer(public_key)?,
+                    Command::DisconnectPeer { public_key } => api.disconnect_peer(public_key)?,
+                    Command::OpenChannel {
+                        public_key,
+                        sats: satoshis,
+                        push_msat,
+                        announce,
+                        fee_rate,
+                    } => api.open_channel(public_key, satoshis, push_msat, announce, fee_rate)?,
+                    Command::SetChannelFee {
+                        id,
+                        base_fee,
+                        ppm_fee,
+                    } => api.set_channel_fee(id, base_fee, ppm_fee)?,
+                    Command::CloseChannel { id } => api.close_channel(id)?,
+                    Command::NetworkNodes { id } => api.list_network_nodes(id)?,
+                    Command::NetworkChannels { id } => api.list_network_channels(id)?,
+                    _ => unreachable!(),
+                }
+            } else {
+                anyhow::bail!("`target`, `cert-path`, `macaroon-path` are required")
+            }
+        }
+        Command::SystemInfo { inline } => {
```

Comment:
You can put this match arm at the top and just use underscore to match everything else.